src/ve-utils/application/Application.service.ts

```diff
@@ -68,70 +66,87 @@ export class ApplicationService {
+                })
+                .then((result) => {
+                    if (result === null) {
+                        this.updateSettings(projectId, refId, null).then(resolve, reject)
+                    } else resolve(result)
+                }, reject)
+        })
     }
 
-    public updatePins = (
+    public createSettings = (
         projectId: string,
         refId: string,
-        prefs: VePreferencesObject
-    ): VePromise<VePreferencesObject, ProjectsResponse> => {
-        const deferred = this.$q.defer<VePreferencesObject>()
-        const cacheKey = this.apiSvc.makeCacheKey({ projectId, refId }, '', false, 'preferences')
-        this.projectSvc.getRef(projectId, refId).then(
-            (ref) => {
-                if (prefs.pinnedIds) {
-                    ref.pinnedIds = prefs.pinnedIds
-                    this.projectSvc.updateRef(ref, projectId).then(
-                        (result: VeRefObject) => {
-                            prefs.pinnedIds = result.pinnedIds
-                            this.cacheSvc.put(cacheKey, prefs)
-                            deferred.resolve(this.cacheSvc.get(cacheKey))
-                        },
-                        (reason) => {
-                            deferred.reject(reason)
-                        }
-                    )
-                } else {
-                    deferred.resolve(prefs)
-                }
-            },
-            (reason) => {
-                deferred.reject(reason)
+        settingsOb?: ProjectSettingsObject
+    ): VePromise<ProjectSettingsObject> => {
+        if (!settingsOb) {
+            settingsOb = {
+                id: '_hidden_' + projectId + '_settings',
+                name: 'View Editor Project Settings',
+                _projectId: projectId,
+                _refId: refId,
+                ownerId: 'holding_bin_' + projectId,
+                type: 'Class',
             }
-        )
-        return deferred.promise
+        }
+        return this.elementSvc.createElement<ProjectSettingsObject>({
+            elementId: '_hidden_' + projectId + '_settings',
+            projectId,
+            refId,
+            elements: [settingsOb],
+        })
+    }
+
+    public updateSettings = (
+        projectId: string,
+        refId: string,
+        settingsOb: ProjectSettingsObject
+    ): VePromise<ProjectSettingsObject> => {
+        return new this.$q<ProjectSettingsObject>((resolve, reject) => {
+            this.elementSvc
+                .getElement<ProjectSettingsObject>({
+                    projectId,
+                    refId,
+                    elementId: '_hidden_' + projectId + '_settings',
+                })
+                .then((result) => {
+                    if (!result) {
+                        this.createSettings(projectId, refId, settingsOb).then(resolve, reject)
+                        return
+                    }
+
+                    if (
+                        settingsOb.pinnedIds &&
```

Comment:
pinnedIds cannot be an object due to the cameo elastic index schema for names that end with Ids/Id, used pinned

Reply:
Okay!